components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+        } catch (ParseException e) {
+            LOG.error("Error occurred while parsing JWT ID token provided by UAEPass: ", e);
+        }
+        Map<String,Object> userInfoJwtAttributes= buildJSON(jwtAttributeSet);
+        return userInfoJwtAttributes;
+    }
+
+
+    /**
+     * Request the access token - Create a request to access token endpoint of the external IdP.
+     *
+     * @param context
+     * @param authzResponse
+     * @return OAuthClientRequest
+     * @throws AuthenticationFailedException
+     */
+    public OAuthClientRequest getAccessTokenRequest(AuthenticationContext context, OAuthAuthzResponse authzResponse)
+            throws UAEPassAuthnFailedException {
+
+        Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+
+        OAuthClientRequest accessTokenRequest = null;
+        try {
+            String envUAEPass = getUAEPassEnvironment(context);
+
+            String clientId = authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.CLIENT_ID);
+            String clientSecret = authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.CLIENT_SECRET);
+            String tokenEndPoint = getTokenUrl(envUAEPass);
+            String callbackUrl = authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.CALLBACK_URL);
+
+            accessTokenRequest = OAuthClientRequest
+                    .tokenLocation(tokenEndPoint)
+                    .setGrantType(GrantType.AUTHORIZATION_CODE)
+                    .setClientId(clientId).setClientSecret(clientSecret)
+                    .setRedirectURI(callbackUrl)
+                    .setCode(authzResponse.getCode())
+                    .buildBodyMessage();
+
+            if (accessTokenRequest != null) {
+                String serverURL = ServiceURLBuilder.create().build().getAbsolutePublicURL();
+                accessTokenRequest.addHeader(UAEPassAuthenticatorConstants.UAE.HTTP_ORIGIN_HEADER, serverURL);
+            }
+
+        } catch (OAuthSystemException e) {
+            LOG.error("Unable to build the request with request's body attributes.");
+            throw new UAEPassAuthnFailedException("Exception while building access token request " +
+                    "with the request body", e);
+        } catch (URLBuilderException e) {
+            if(LOG.isDebugEnabled()) {
+                LOG.error("Unable to identify common-auth URL on browser");
+            }
+            throw new UAEPassAuthnFailedException("Error occurred while extracting the absolute public URL from browser.", e);
+        }
+        return accessTokenRequest;
+    }
+
+
+    /**
+     * Returns the OAuth type response to the back channel.
+     *
+     * @param oAuthClient
+     * @param accessRequest
+     * @return OAuthClientResponse
+     * @throws AuthenticationFailedException
+     */
+    public OAuthClientResponse getOauthResponse(OAuthClient oAuthClient, OAuthClientRequest accessRequest)
```

Comment:
```suggestion
    public OAuthClientResponse getOAuthResponse(OAuthClient oAuthClient, OAuthClientRequest accessRequest)
```
Let's keep one consistent formatting for OAuth

Reply:
Refactored and fixed.